src/main/java/org/opensearch/security/privileges/PrivilegesEvaluatorResponse.java

```diff
@@ -79,11 +161,50 @@ public boolean isPending() {
 
     @Override
     public String toString() {
-        return "PrivEvalResponse [allowed=" + allowed + ", missingPrivileges=" + missingPrivileges + "]";
+        return "PrivEvalResponse [\nallowed="
+            + allowed
+            + ",\nonlyAllowedForIndices="
+            + onlyAllowedForIndices
+            + ",\n"
+            + (indexToActionCheckTable != null ? indexToActionCheckTable.toTableString("ok", "MISSING") : "")
+            + "]";
+    }
+
+    public static PrivilegesEvaluatorResponse ok() {
+        PrivilegesEvaluatorResponse response = new PrivilegesEvaluatorResponse();
+        response.allowed = true;
+        return response;
+    }
+
+    public static PrivilegesEvaluatorResponse partiallyOk(
+        Set<String> availableIndices,
+        CheckTable<String, String> indexToActionCheckTable,
+        PrivilegesEvaluationContext context
+    ) {
+        PrivilegesEvaluatorResponse response = new PrivilegesEvaluatorResponse();
+        response.onlyAllowedForIndices = ImmutableSet.copyOf(availableIndices);
+        response.indexToActionCheckTable = indexToActionCheckTable;
+        return response;
+    }
+
+    public static PrivilegesEvaluatorResponse insufficient(String missingPrivilege, PrivilegesEvaluationContext context) {
+        PrivilegesEvaluatorResponse response = new PrivilegesEvaluatorResponse();
+        response.indexToActionCheckTable = CheckTable.create(ImmutableSet.of("_"), ImmutableSet.of(missingPrivilege));
+        return response;
+    }
+
+    public static PrivilegesEvaluatorResponse insufficient(
```

Comment:
If the second arg is not needed here can it be removed?

Reply:
Done in https://github.com/opensearch-project/security/pull/4380/commits/c474bf7049e85126beef41ad6483fb1e8c3e9b73